mantidproject/mantid
# Intial layout and startup scripts for workbench package

**Description of work**

Adds the `workbench` python package plus minimal start up code to give a functioning application. 

**To test:**

* You will need the Qt 5 libraries:
 * Windows: nothing to do
 * Linux: Update the relevant `mantid-developer` package for your platform  
 * OSX: Use `brew` to install `qt`, `sip` & `pyqt`. See instructions [here](https://github.com/mantidproject/mantid/wiki/Build-environment-setup-on-OS-X-10.12-Sierra)  
* Checkout the branch and run `cmake -DENABLE_WORKBENCH=ON .`
* Build the `workbench` target
  * it should create an egg link in the bin directory to the source code without copying it
* ctest -R workbench should run the tests
* `bin/workbench` (Linux/OSX) or `bin/CONFIG/workbench.exe` (windows) should start the application (CTRL+Q) to quit.

Fixes #20381 

**Release Notes** 

*Does not need to be in the release notes as there is nothing to shout about yet.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
